### PR TITLE
extend joda's DateTime to support micro second in order to hold Timestamp

### PR DESCRIPTION
**Pull request by marsishandsome**
let joda's DateTime hold micro second

close https://github.com/pingcap/tispark/issues/770

### CODE REVIEW

**Comment by marsishandsome** [comment]
@birdstorm @zhexuany PTAL

File: core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala [hunk 1]
```diff
@@ -30,20 +30,20 @@ class ToDateTimeSuite extends BaseDataSourceTest("test_data_type_convert_to_date
...
     val readB: java.sql.Timestamp = java.sql.Timestamp.valueOf("1990-01-01 01:01:01.999")
-    val readC: java.sql.Timestamp = java.sql.Timestamp.valueOf("1995-05-01 21:21:21.6")
+    val readC: java.sql.Timestamp = java.sql.Timestamp.valueOf("1995-05-01 21:21:21.666666")
 
     readRow1 = Row(1, null, null, null)
     readRow2 = Row(2, readA, readB, readC)
   }
 
-  test("Test Convert from java.lang.Long to DATETIME") {
+  ignore("Test Convert from java.lang.Long to DATETIME") {
```
Comment (birdstorm) [inline core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala]: why is this test ignored?

File: tikv-client/src/main/java/com/pingcap/tikv/types/DateTimeType.java [hunk 2]
```diff
@@ -56,7 +57,8 @@ protected Object doConvertToTiDBType(Object value)
       throws ConvertNotSupportException {
     java.sql.Timestamp result;
     if (value instanceof Long) {
-      result = new java.sql.Timestamp((Long) value);
+      throw new ConvertNotSupportException(value.getClass().getName(), this.getClass().getName());
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/types/DateTimeType.java]: I think this logic should be something like `someExtendedTimestamp.toTimestamp()`?

File: core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala (hunk 1 above)
Comment (marsishandsome) [reply core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala]: 1. do not sure what long means? maybe microseconds or milliseconds from 1970-01-01
2. should timezone be considered?

will support after investigate

Comment (zhexuany) [reply core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala]: what's the behavior in tidb side?

Comment (marsishandsome) [reply core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala]: after investigate mysql do not support insert long type to DATETIME

Comment (birdstorm) [reply core/src/test/scala/com/pingcap/tispark/convert/ToDateTimeSuite.scala]: I believe we can delete this test or check if it is not supported.

**Pull request merged**